### PR TITLE
Added cmdlets protection job for NAS and migrating Copy-CohesityVMwareVM to script based

### PR DESCRIPTION
**Pull request by bijayprasadtiwari**

### CODE REVIEW

File: src/Cohesity.Powershell/Scripts/Clone/Copy-CohesityVMwareVM.ps1 [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        Clones the specified VMware virtual machine.
+        .DESCRIPTION
+        Clones the specified VMware virtual machine. The cmdlet can copy VM from remote cluster as well.
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Copy-CohesityVMwareVM -TaskName "test-clone-task" -SourceId 883 -TargetViewName "test-vm-datastore" -JobId 49402 -VmNamePrefix "clone-" -DisableNetwork -PoweredOn -ResourcePoolId 893
+        Clones the VMware virtual machine with the given source id using the latest run of job id 49402.
```
Comment (pyashish) [inline src/Cohesity.Powershell/Scripts/Clone/Copy-CohesityVMwareVM.ps1:13]: Is this comment or part of the cmdlet ?

File: docs/cmdlets-reference/copy-cohesityvmwarevm.md [hunk 2]
```diff
@@ -34,9 +44,9 @@ Type: String
 Parameter Sets: (All)
 Aliases:
 
-Required: True
+Required: False
```
[the remark this answers is not in the document]
Comment (bijayprasadtiwari) [reply docs/cmdlets-reference/copy-cohesityvmwarevm.md:47]: We are not removing any parameter, in this case TaskName has been made non-mandatory parameter.

File: src/Cohesity.Powershell/Scripts/Clone/Copy-CohesityVMwareVM.ps1 (hunk 1 above)
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/Clone/Copy-CohesityVMwareVM.ps1:13]: The cmdlet examples followed by the comments.

File: docs/cmdlets-reference/new-cohesitynasprotectionjob.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+### EXAMPLE 1
+```
+New-CohesityNASProtectionJob -Name job-nas -PolicyName Bronze -StorageDomainName DefaultStorageDomain -SourceName "10.14.31.60:/view1"
+```
+
+Creating job for a NFS mount NAS source.
+
+### EXAMPLE 2
+```
+New-CohesityNASProtectionJob -Name job-smb1 -PolicyName Bronze -StorageDomainName DefaultStorageDomain -SourceName "\\10.14.31.156\view3" -TimeZone "Asia/Kolkata"
```
[the remark this answers is not in the document]
Comment (bijayprasadtiwari) [reply docs/cmdlets-reference/new-cohesitynasprotectionjob.md:27]: Done

File: src/Cohesity.Powershell/Scripts/Clone/Copy-CohesityVMwareVM.ps1 [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+                $cohesityUrl = $cohesityCluster + '/irisservices/api/v1/public/restore/clone'
+            }
+            $payloadJson = $cloneRequest | ConvertTo-Json -Depth 100
+            $cohesityHeaders = @{'Authorization' = 'Bearer ' + $cohesityToken }
+            $resp = Invoke-RestApi -Method Post -Uri $cohesityUrl -Headers $cohesityHeaders -Body $payloadJson
+            if ($resp) {
+                $resp
+            }
+            else {
+                $errorMsg = "VMwareVM : Failed to copy."
```
[the remark this answers is not in the document]
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/Clone/Copy-CohesityVMwareVM.ps1:274]: Fixed

File: src/Cohesity.Powershell/Scripts/ProtectionJob/New-CohesityNASProtectionJob.ps1 [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+            if ("System.Array" -eq $protectionSourceObject.GetType().BaseType.ToString()) {
+                # In case the name of the object is same across different registered sources
+                Write-Output "There are multiple objects found for the search item, selecting the first item as source"
+                $protectionSourceObject = $protectionSourceObject[0]
+            }
+            $payload = @{
+                name           = $Name
+                policyId       = $protectionPolicyObject.Id
+                _policyName    = $protectionPolicyObject.Name
```
[the remark this answers is not in the document]
Comment (bijayprasadtiwari) [reply src/Cohesity.Powershell/Scripts/ProtectionJob/New-CohesityNASProtectionJob.ps1:82]: Fixed

**Pull request merged**